src/meta/uchange.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2020 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner
+-/
+
+/-!
+# Changing universes of types in meta-code
+
+This file defines the meta type `uchange (α : Type v) : Type u`, which
+permits us to change the universe of a type analogously to `ulift`.
+However since `uchange` is meta, it can both lift and lower the universe.
+
+The implementation of `uchange` is efficient. Both `uchange.up` and
+`uchange.down` compile to no-ops.
+-/
+
+universes u v
+
+/--
+`unchecked_cast a : β` performs of an unchecked cast of `(a : α)` to `β`.
+
+Unlike `unchecked_cast`, it can cast across universes. The VM implementation
+is guaranteed to be the identity.
+-/
+@[inline, irreducible]
+meta def unchecked_cast' {α : Sort u} {β : Sort v} (a : α) : β :=
+plift.down $ @cast (α → β → plift β) (β → α → plift β) undefined (λ _ a, plift.up a)
```

Comment:
Unfortunately, Lean is not clever enough to accept this definition without the `plift` because it can't prove that `imax u v = imax v (imax u v)`.